index.js

```diff
@@ -53,10 +66,7 @@ Equivalency._collapseRules = function(rules) {
       case 'RemoveRule':
       case 'MapRule': {
         if (!matters) {
-          collapsedMap = new Map([
-            ...collapsedMap.entries(),
-            ...rule.entries(),
-          ]);
+          concatMaps(collapsedMap, rule);
```

Comment:
This way of merging maps is faster b/c it avoids allocations for new Map instances.